polygeist/include/mlir/Dialect/Polygeist/Analysis/ReachingDefinitionAnalysis.h

```diff
@@ -43,44 +93,57 @@ class ReachingDefinition : public dataflow::AbstractDenseLattice {
   friend raw_ostream &operator<<(raw_ostream &, const ReachingDefinition &);
 
 public:
-  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(ReachingDefinition)
-
   using AbstractDenseLattice::AbstractDenseLattice;
-  using ModifiersTy = SetVector<Operation *, SmallVector<Operation *>,
-                                SmallPtrSet<Operation *, 2>>;
+  using DefinitionPtr = std::shared_ptr<Definition>;
+
+  struct CompareDefinition {
+    bool operator()(DefinitionPtr lhs, DefinitionPtr rhs) const {
+      return (lhs && rhs) ? std::less<Definition>{}(*lhs, *rhs)
+                          : std::less<DefinitionPtr>{}(lhs, rhs);
+    }
+  };
+  using ModifiersTy = std::set<DefinitionPtr, CompareDefinition>;
```

Comment:
Question: Why do we need a `shared_ptr` here? Can we just insert the `Definition`?

Reply:
We can, in this case the Definition class is not too big.